coffee/plan.py

```diff
@@ -176,6 +178,8 @@ def _generate_cpu_code(self, kernel, **kwargs):
         else:
             params = opts
 
+        update_citations(params)
```

Comment:
The only bad thing about this approach is that if we hit the firedrake kernel cache, coffee is never called, and so update_citations(params) is never called.